apps/crn-server/test/data-providers/contentful/analytics/collaboration.analytics.data-provider.test.ts

```diff
@@ -83,3 +93,455 @@ describe('collaboration', () => {
+      contentfulGraphqlClientMock.request.mockResolvedValue(
+        getTeamCollaborationQuery(),
+      );
+
+      await analyticsDataProvider.fetchTeamCollaboration({});
+
+      expect(contentfulGraphqlClientMock.request.mock.calls).toEqual([
+        [
+          FETCH_TEAM_COLLABORATION,
+          expect.objectContaining({
+            limit: 5,
+            skip: 0,
+          }),
+        ],
+        [
+          FETCH_TEAM_COLLABORATION,
+          expect.objectContaining({
+            limit: 5,
+            skip: 5,
+          }),
+        ],
+      ]);
+    });
+  });
+
+  test('Should fetch the list of team collaborations from Contentful GraphQl', async () => {
+    const result =
+      await analyticsDataProviderMockGraphql.fetchTeamCollaboration({
+        take: 1,
+      });
+
+    expect(result).toMatchObject({
+      total: 1,
+      items: [getTeamCollaborationDataObject()],
+    });
+  });
+
+  test('Should return an empty result when the client returns an empty list', async () => {
+    contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+      teamsCollection: {
+        items: [],
+        total: 0,
+      },
+    });
+
+    const result = await analyticsDataProvider.fetchTeamCollaboration({});
+
+    expect(result).toEqual({
+      total: 0,
+      items: [],
+    });
+  });
+
+  test('Should return an empty result when the client returns teamsCollection as null', async () => {
+    contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+      teamsCollection: null,
+    });
+
+    const result = await analyticsDataProvider.fetchTeamCollaboration({});
+
+    expect(result).toEqual({
+      total: 0,
+      items: [],
+    });
+  });
+
+  test('Should return the count of outputs as zero when the client returns researchOutputCollection as null', async () => {
+    const graphqlResponse = getTeamCollaborationQuery();
+    graphqlResponse.teamsCollection!.items[0]!.linkedFrom!.researchOutputsCollection =
+      null;
+    contentfulGraphqlClientMock.request.mockResolvedValueOnce(graphqlResponse);
+
+    const result = await analyticsDataProvider.fetchTeamCollaboration({
+      take: 1,
+    });
+
+    expect(result).toEqual({
+      total: 1,
+      items: [
+        expect.objectContaining({
+          outputsCoProducedWithin: {
+            Article: 0,
+            Bioinformatics: 0,
+            Dataset: 0,
+            'Lab Resource': 0,
+            Protocol: 0,
+          },
+          outputsCoProducedAcross: {
+            byDocumentType: {
+              Article: 0,
+              Bioinformatics: 0,
+              Dataset: 0,
+              'Lab Resource': 0,
+              Protocol: 0,
+            },
+            byTeam: [],
+          },
+        }),
+      ],
+    });
+  });
+  describe('within team', () => {
+    test('Should return a count of 2 if client has two outputs of the same document type with multiple labs', async () => {
```

Comment:
it seems to have three outputs of type 'Article' though, no?

Reply:
No, it's 2. One has its addedDate outside the 30 day range